### PR TITLE
Object' is a special class name (rename Object to BaseObject)

### PR DESCRIPTION
**Pull request by djoudi**
Object' is a special class name (rename Object to BaseObject)

### CODE REVIEW

**Comment by djoudi** [comment]
Object' is a special class name (rename Object to BaseObject)

**Comment by djoudi** [comment]
Object is now a reserved class name in PHP 7.2

File: composer.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-    "name": "php-xapi/model",
+    "name": "djoudi/model",
```
Comment (xabbuh) [inline composer.json]: please revert this change

**Comment by xabbuh** [comment]
I would have expected changes in other classes that extend the base `Object` class.

File: composer.json (hunk 1 above)
Comment (djoudi) [reply composer.json]: Ok now

**Review by djoudi — COMMENTED** [review]
update

**Review by xabbuh — CHANGES_REQUESTED** [review]
This still needs changes in the other classes that still reference the old name (you can take a look at the failing Travis CI build).

**Comment by djoudi** [comment]
waiting accepted ...

**Comment by xabbuh** [comment]
Let's use `StatementObject` as the class name and make this change consistently throughout the code base.

**Comment by djoudi** [comment]
check now

**Comment by djoudi** [comment]
waiting accepted ...

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,4 @@
 /composer.lock
 /phpunit.xml
-/vendor
+/vendor/
+/.idea/
```
Comment (xabbuh) [inline .gitignore:4]: Please do not add this line. Editors are user specific and should therefore be added to your global `.gitignore` file if any

File: .idea/composerJson.xml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (xabbuh) [inline .idea/composerJson.xml]: please remove this file and all other files from the `.idea` directory

**Review by xabbuh — CHANGES_REQUESTED** [review]

**Pull request closed without merging**